### PR TITLE
Manipulators

### PR DESCRIPTION
**Pull request by georgia-martinez**
For toggling the manipulators on and off with the ps5 controller

### CODE REVIEW

File: pi/manipulators/launch/manip_launch.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import os
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from ament_index_python.packages import get_package_share_directory
+
+
+def generate_launch_description():
+    ld = LaunchDescription()
```
Comment (InvincibleRMC) [inline pi/manipulators/launch/manip_launch.py:8]: Optional: construct LaunchDescription at end of generate_launch_description with a list of actions
We do that method in all of our other launch files

```python
manip_node = ...

return LaunchDescription([manip_node])
```

File: pi/manipulators/manipulators/i2c.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+link to permanant solution
+https://unix.stackexchange.com/questions/147494/how-can-i-set-device-rw-permissions-permanently-on-raspbian
+"""
+
+from rclpy.node import Node
+from manipulators.tca9555 import TCA9555
+
+
+class I2CTest(Node):
+    pass
```
Comment (InvincibleRMC) [inline pi/manipulators/manipulators/i2c.py:15]: Is this class necessary anymore?

File: pi/manipulators/manipulators/i2c.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
+"""
+Need to run this command.
+
+sudo chmod a+rw /dev/i2c-*
+link to permanant solution
+https://unix.stackexchange.com/questions/147494/how-can-i-set-device-rw-permissions-permanently-on-raspbian
+"""
```
Comment (InvincibleRMC) [inline pi/manipulators/manipulators/i2c.py:8]: Add note that we actually implemented the permanent solution

File: pi/manipulators/manipulators/i2c.py
Comment (InvincibleRMC) [inline pi/manipulators/manipulators/i2c.py:1]: Rename this file to something like `manipulator_test.py`

File: pi/manipulators/package.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>manipulators</name>
+  <version>0.0.0</version>
+  <description>A generic package for showing how to do manipulators.</description>
+  <maintainer email="case@case.edu">CWRUBotics 2022-2023</maintainer>
```
Comment (InvincibleRMC) [inline pi/manipulators/package.xml:7]: Update to use at least the cwrubotix address & have a better description

File: pi/manipulators/setup.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        ('share/' + package_name, ['package.xml']),
+        # Include all launch files.
+        (os.path.join('share', package_name, 'launch'),
+         glob('launch/*launch.[pxy][yma]*')),
+        (os.path.join('share', package_name, 'config'),
+         glob('config/*')),
+        (os.path.join('lib', f'python{major_num}.{minor_num}', 'site-packages', package_name),
+         glob(os.path.join('TCA9555', 'tca9555', 'tca9555.py'))),
+    ],
+    install_requires=['setuptools', 'wiringpi', 'bitstring'],
```
Comment (InvincibleRMC) [inline pi/manipulators/setup.py:28]: If wiringpi is still necessary, it should also be in the package.xml
If it's not, remove it here

File: pi/manipulators/setup.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    zip_safe=True,
+    maintainer='taz',
+    maintainer_email='tzupfer@gmail.com',
+    description='Code for manipulators',
```
Comment (InvincibleRMC) [inline pi/manipulators/setup.py:32]: Better description

File: surface/task_selector/task_selector/manual_control_node.py [hunk 7]
```diff
@@ -114,6 +149,28 @@ def cancel_callback(self, goal_handle: ServerGoalHandle):
         self._passing = False
         return CancelResponse.ACCEPT
 
+    def manip_callback(self, msg: Joy):
```
Comment (InvincibleRMC) [inline surface/task_selector/task_selector/manual_control_node.py:152]: Add typing to variables in this method

**Review by InvincibleRMC — CHANGES_REQUESTED** [review]

**Review by benjaminwp18 — APPROVED** [review]

**Review by InvincibleRMC — APPROVED** [review]

**Pull request merged**